TANMEDIAGROUP/SYNDEX_BN
# google -auth

* * *

### What does this PR do?

This pull request reduces some redundancies in the codebase, adds a Google Auth route, provides more flexibility for wrappers, and adds salt for extra password security.

### Description of Task to be completed?

The key changes in this PR include:

#### Tasks

- [x] Removed duplicate code and consolidated common functionality
- [x] Implemented a new Google Auth route for user authentication
- [x] Added a wrapper component that can be more easily customized
- [x] Introduced salt to the password hashing process for improved security

### How to test

- Run the application locally
- Verify that the redundant code has been removed and the codebase is more concise
- Test the new Google Auth route by attempting to log in with a Google account
- Observe the new wrapper component and try modifying its appearance and behavior
- Inspect the password hashing code to ensure salt has been added

### Any background context you want to provide?

The redundancy reduction was necessary to improve maintainability and reduce technical debt. The Google Auth route was added to provide users with more options for authentication. The wrapper component was updated to give designers and developers more flexibility in customizing the user interface. Finally, the addition of salt to the password hashing process enhances the overall security of the application.

### What are the relevant Github Issues if any?

null

#### Tracker ID:

Trello card #21324

* * *
